File: Top-Shop-UI/src/app/models/vendorModel/PaymentInformation.ts
export class PaymentInformation {
   id:number;
   nameOnCard:string;
   secCode:string;
   expDate:string;
   cardType:string;
   cardNumber:string;


  constructor(nameOnCard: string, secCode: string, expDate: string, cardType: string, cardNumber: string) {
    this.nameOnCard = nameOnCard;
    this.secCode = secCode;
    this.expDate = expDate;
    this.cardType = cardType;
    this.cardNumber = cardNumber;
  }
}

File: Back-End/user/src/main/java/com/top/shop/user/domain/PaymentInformation.java
package com.top.shop.user.domain;

import com.sun.istack.NotNull;
import com.top.shop.user.util.CardType;
import lombok.Data;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;
/**
 * @author Yome Mengistu
 */
@Entity
@Data
public class PaymentInformation {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @NotNull
    private String nameOnCard;
    @NotNull
    @Size(min = 3,max = 3)
    private String secCode;
    @NotNull
    private String expDate;
    @NotNull
    private CardType cardType;
    @NotNull
    @Size(min = 16,max = 16)
    private String cardNumber;

    }
